File: frontend/src/cube/sidecoord.ts
import { Side } from "./side";

export default class SideCoord {
    side: Side;
    i: number;
    j: number;
    constructor(side: Side, i: number, j: number) {
        this.side = side;
        this.i = i;
        this.j = j;
    }
}

File: backend/src/config/generatorvisitor/sidecoord.go
package generatorvisitor

import (
	"github.com/taskat/rubiks-cube/src/models/coord"
	"github.com/taskat/rubiks-cube/src/models/parameters"
)

type sideCoord struct {
	side parameters.Side
	coord.Coord
}

func newSideCoord(side parameters.Side, row, col int) sideCoord {
	return sideCoord{side, coord.NewCoord(row, col)}
}
